test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,395 @@
+            System.err.print("Events triggered are: ");
+            for (int j = 0; j < events.size();
+                System.err.print(events.get(j) + "; "), j++);
+            System.err.println();
+            throw new RuntimeException(
+                "FAIL: ComponentEvent triggered when frame is set to normal state");
+        }
+    }
+
+    private static void doTestForComponent(int i)
+        throws InvocationTargetException, InterruptedException {
+        Component currentComponent = components[i];
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(false);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentHidden) {
+            throw new RuntimeException(
+                "FAIL: ComponentHidden not triggered for "
+                    + currentComponent.getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(false);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (componentHidden) {
+            throw new RuntimeException(
+                "FAIL: ComponentHidden triggered when setVisible(false) "
+                    + "called for a hidden " + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(true);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentShown) {
+            throw new RuntimeException("FAIL: ComponentShown not triggered for "
+                + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setVisible(true);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (componentShown) {
+            throw new RuntimeException(
+                "FAIL: ComponentShown triggered when setVisible(true) "
+                    + "called for a shown " + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setLocation(currentComponent.getLocation().x + 1,
+                currentComponent.getLocation().y);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentMoved) {
+            throw new RuntimeException("FAIL: ComponentMoved not triggered for "
+                + components[i].getClass());
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            currentComponent.setSize(currentComponent.getSize().width + 1,
+                currentComponent.getSize().height);
+            resetFrame();
+        });
+
+        robot.delay(DELAY);
+        if (!componentResized) {
+            throw new RuntimeException(
+                "FAIL: ComponentResized not triggered for "
+                    + components[i].getClass());
```

Comment:
Choose one style… At line 267, you pass the message on the same line; here you wrap the message to next line immediately.

It's more common to keep the first part of the string on the same line and wrap the string as necessary.